Libraries/src/net/relinc/libraries/sample/Sample.java

```diff
@@ -203,25 +220,38 @@ public boolean writeBarSetupToSampleFile(String sampleZipPath) {
 
 		
 	}
-	
+
 	private String getStringForFileWriting() {
-		return getCommonString() + getSpecificString() + (strikerBar.isValid() ? "StrikerBar" + delimiter + strikerBar.getStringForFile() : "");
+		JSONObject jsonObject = getCommonJSON();
+
+		if(strikerBar.isValid()) {
+			jsonObject.put("StrikerBar", strikerBar.getStringForFile());
+		}
+		addSpecificToJSONObject(jsonObject);
```

Comment:
So this is where I'd merge the JSON instead of using a side effect. Unfortunately this library doesn't make it easy, so maybe it's not better after all... https://stackoverflow.com/questions/2403132/merge-concat-multiple-jsonobjects

Reply:
I tried the for loop. JSONObject.getNames() doesn't exist. I found that JSONObject has a putAll() method that can be used to "merge" but it still uses JSONObject as an argument.


		private String getStringForFileWriting() {
		JSONObject jsonObject = getCommonJSON();

		if(strikerBar.isValid()) {
			jsonObject.put("StrikerBar", strikerBar.getStringForFile());
		}
		JSONObject merge = addSpecificToJSONObject();
		jsonObject.putAll(merge);

		return jsonObject.toString();
	}